src/notification.c

```diff
@@ -1077,6 +1078,10 @@ handle_add_in_thread_func (GTask        *task,
       return;
     }
 
+  desktop_file = xdp_app_info_get_desktop_file (call_data->app_info);
+  if (desktop_file)
```

Comment:
probably we should check here whether `desktop_file == ""`.  Or maybe a follow up MR could make sure that the `desktop_file` is never set to "".

Reply:
Changed it so we never get an empty string or ".desktop" as `desktop_file_id` but in general, it needs to handle not finding the file anyway so this doesn't really change anything.